extension-api/extension-api.ts

```diff
@@ -77,13 +77,21 @@ export interface AccountConfig {
 /***********************************
  * Types for the context objects used by service deployers
  ***********************************/
+
+export interface ServiceRegistry {
+    findDeployerFor(prefix: string, name: string): ServiceDeployer;
+    hasService(prefix: string, name: string): boolean;
+    validPrefixes(): Set<string>;
```

Comment:
Hmm.  I'm gonna actually push a commit that makes these names better.  Gimme a few minutes.